cyences_app_for_splunk/appserver/static/privilege_user_role_editor.js

```diff
@@ -0,0 +1,289 @@
+    function runUserInventoryUpdaterSearch(entries_to_update, callBackFunction){
+        // Splitting array into multiple queries in case bulk update contains many results, it may result into failure
+        const chunkSize = 500;
+        let searchesCompleted = {};
+        for (let i = 0; i < entries_to_update.length; i += chunkSize) {
+            const chunk = entries_to_update.slice(i, i + chunkSize);
+
+            let user_role = chunk[0].user_role;
+            let searchQuery = '';
+            for(let i=0; i<chunk.length; i++){
+                let entry = chunk[i];
+                if (i==0){
+                  searchQuery = `| inputlookup cs_user_inventory | search uuid IN (`;
+                }
+                searchQuery += `"${entry.uuid}",`;
+            }
+            searchQuery = searchQuery.slice(0,-1); 
+
+            if (user_role==DEFAULT_USER_ROLE){
+                searchQuery += `) | eval _key=uuid, privilege_user_role=null() | outputlookup cs_user_inventory append=true`;
+            }
+            else {
+                searchQuery += `) | eval _key=uuid, privilege_user_role="${user_role}" | outputlookup cs_user_inventory append=true`;
+            }
+            console.log("Search Query for user inventory update: ", searchQuery);
+
+            searchesCompleted[i] = false;
+
+            new SplunkCommonUtilities.VSearchManagerUtility(
+                function(results){
+                    if (results != null) {
+                        // TODO - read through the output of the results and validate the custom command was successful.
+                        searchesCompleted[i] = true;
+                    }
+                },
+                function(errorProperties){
+                    alert("Unable to update the user entry in user inventory.");
+                }
+            ).searchByQuery(searchQuery);
+        }
+
+        SplunkCommonUtilities.vWaitUntil(
+            function(){
+                for(let key in searchesCompleted){
+                    if (searchesCompleted[key] === false){
+                        return false;
+                    }
+                }
+                return true;
+            },
+            function(){
+                restartUserEventSearch();
+                if(callBackFunction != undefined){
+                    callBackFunction();
+                }
+            }
+        )
+    }
+
+    // create the popup dialoge box when user wants to edit the entries.
+    function handlerUserInventoryEdit(handlerObj, data){
+        console.log("user_event_edit handler.");
+
+        let modal_title = "Privilege User Role";
+        let nr_user_events = data.user_event_uuids.length;
+        let user_event_uuids_string = data.user_event_uuids.join(':');
+
+        var edit_panel = '' +
+            '<div class="modal fade modal-wide shared-alertcontrols-dialogs-editdialog in" id="edit_panel">' +
```

Comment:
As this dialog box html is pretty much constant please add the whole modal as part of a HTML separately only once when the JS loads.

And then from this function just show/hide the modal after changing necessary values.